identity/src/keypair.rs

```diff
@@ -167,34 +159,37 @@ impl Keypair {
     #[cfg(all(feature = "rsa", not(target_arch = "wasm32")))]
     pub fn rsa_from_pkcs8(pkcs8_der: &mut [u8]) -> Result<Keypair, DecodingError> {
         #[allow(deprecated)]
```

Comment:
I recall in my pr attempt, it was needed unless we switch to using ` rsa::Keypair::try_decode_pkcs8`

Reply:
Ah right, these probably didn't surface when we landed https://github.com/libp2p/rust-libp2p/pull/3775 because of the `allow(deprecated)` we had on here for the enum variant.

Mind fixing those up as well? It shouldn't cause too much noise and landing an additional PR after seems a bit unnecessary!

Reply:
@thomaseizinger Sure, done, no more `allow(deprecated)` in `keypair.rs`